### PR TITLE
Dls 8632 nav fix imports

### PR DESCRIPTION
**Pull request by GinnyThomas**

### CODE REVIEW

File: app/navigation/Navigator.scala [hunk 1]
```diff
@@ -122,11 +122,24 @@ class Navigator @Inject()() {
...
       case true => routes.HowManyImportsController.onPageLoad(mode)
-      case false if mode == NormalMode => routes.StartDateController.onPageLoad(mode)
-      case _ => routes.CheckYourAnswersController.onPageLoad
+      case false =>
+        if (navigateToStartDate(userAnswers)) {
+          routes.StartDateController.onPageLoad(mode)
+        } else {
+          contactDetailsOrDoNotRegister(userAnswers, mode)
+        }
+      case _ =>  routes.CheckYourAnswersController.onPageLoad
```
Comment (Chrissystk) [inline app/navigation/Navigator.scala:131]: Case _ is not required also the if can be put in the case i.e

```
case false if navigateToStartDate(userAnswers) => routes.StartDateController.onPageLoad(mode)
case false => contactDetailsOrDoNotRegister(userAnswers, mode)
```

File: app/controllers/actions/RequiredUserAnswers.scala [hunk 2]
```diff
@@ -49,66 +49,70 @@ class RequiredUserAnswers @Inject()(genericLogger: GenericLogger)(implicit val e
...
+              case Nil => !previousPageAnswer.isDefined
+              case _ => !previousPageAnswer.exists(i => previousPage.previousPageAnswerRequired.contains(i))
+            }
           }
-          if(userAnswersMatched.contains(true) && userAnswersMatched.contains(false)) {
+          if (userAnswersMatched.contains(true) && userAnswersMatched.contains(false)) {
             true
           } else {
             !userAnswersMatched.contains(false)
           }
```
Comment (Chrissystk) [inline app/controllers/actions/RequiredUserAnswers.scala:70]: I'm not sure what the purpose of this if statement is the first part is returning true if the userAnswersMatched contains both true and false and the else element is returning true if the userAnswersMatched contains false. So whether the userAnswersMatched contains true doesn't seem to matter and if ```
(userAnswersMatched.contains(true) && userAnswersMatched.contains(false)) {
            true
          } else {
            !userAnswersMatched.contains(false)
          }
``` could be replaced with `!userAnswersMatched.contains(false)`

**Review by Chrissystk — CHANGES_REQUESTED** [review]

File: app/controllers/actions/RequiredUserAnswers.scala (hunk 2 above)
Comment (ChrisHillCap) [reply app/controllers/actions/RequiredUserAnswers.scala:70]: its if  1 previous answers is answered but the other one isnt, then we do not require the current page to be answered
userAnswersMatched.contains(true) <- previous user answer doesnt exist
userAnswersMatched.contains(false) <- previous user answer does exist

else
if it doesnt contain true, and only contains false (by virtue of the if statement) then we require the current user answer

Comment (ChrisHillCap) [reply app/controllers/actions/RequiredUserAnswers.scala:70]: as its a list

Comment (ChrisHillCap) [reply app/controllers/actions/RequiredUserAnswers.scala:70]: image
brain hurter

File: app/navigation/Navigator.scala (hunk 1 above)
Comment (GinnyThomas) [reply app/navigation/Navigator.scala]: Completed

File: app/controllers/actions/RequiredUserAnswers.scala (hunk 2 above)
Comment (Chrissystk) [reply app/controllers/actions/RequiredUserAnswers.scala]: My point was it is returning true whether you check for if it contains true or not.

Comment (Chrissystk) [reply app/controllers/actions/RequiredUserAnswers.scala]: We can leave it for now though given we are going to be having another story around this.

**Pull request closed without merging**